internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -123,9 +132,12 @@ func New(sess *session.Session) CloudFormation {
 			}))
 		},
 		appStackSet: stackset.New(sess),
-		s3Client:    s3.New(sess),
+		appS3Client: s3.New(sess),
```

Comment:
Is there any reason why previously we can use the env manager role but now we have to use the profile role 🤔

Reply:
We didn't use `s3Client` when we deploy a workload, so the permission issue didn't surface. This field was previously used only by pipeline.

Reply:
Check out the two commits that I pushed on top of your changes: https://github.com/efekarakus/copilot-cli/commits/upload-template-to-s3

I know that today for addons, we use the `defaultSessWithEnvRegion`, but I think that violates the idea that `copilot deploy` only needs to assume the `EnvManagerRole` to deploy a workload.

Instead, 
1. https://github.com/efekarakus/copilot-cli/commit/d22247fa1612d58adbe38504d1c782ea8c4ccfdc is a prototype that shows how we can augment the `EnvManagerRole` with permission to push to the artifact bucket
2. https://github.com/efekarakus/copilot-cli/commit/f8f10894bf60064221e31976ee7993424c25d911 easy peasy, swap the session

I tested the above and it works.

I think `env init` can get away with using `defaultSessWithEnvRegion` because you need to be an admin to create an environment, however to deploy a service users should only need ideally the EnvManagerRole.

Hope this makes sense!